src/instruments/src/ATC/index.tsx

```diff
@@ -0,0 +1,132 @@
+import './style.scss';
+import React, { useEffect, useState } from 'react';
+import { render } from '../Common';
+import { useSimVar, useSplitSimVar } from '../Common/simVars';
+import { useInteractionEvent, useUpdate } from '../Common/hooks';
+
+const getDigitsFromCode = (code: number): number[] => {
+    let codeCopy = code;
+    const digits: number[] = [];
+    while (codeCopy > 0) {
+        digits.push(codeCopy % 10);
+        codeCopy = Math.floor(codeCopy / 10);
+    }
+    digits.reverse();
+    return digits;
+};
+const getBco16FromDigits = (digits: number[]): number => (digits[0] * 4096) + (digits[1] * 256) + (digits[2] * 16) + digits[3];
+
+const PoweredXpdrDisplay = () => {
+    const [currentDigits, setCurrentDigits] = useState([0, 0, 0, 0]);
+    const [currentEnteredDigits, setEnteredDigits] = useState(4);
+    const [doubleClrTimer, setDoubleClrTimer] = useState(-1);
+    const [displayResetTimer, setDisplayResetTimer] = useState(-1);
+    const [ltsTest] = useSimVar('L:XMLVAR_LTS_Test', 'Bool', 250);
+
+    const [transponderCode, setTransponderCode] = useSplitSimVar('TRANSPONDER CODE:1', 'number', 'K:XPNDR_SET', 'Bco16', 250);
+
+    // Check if the transponder code simvar has changed. If it has, and we are not currently editing the code,
+    // set the display to the new code.
+    useEffect(() => {
+        if (currentEnteredDigits === 4) {
+            const newDigits = getDigitsFromCode(transponderCode);
+            if (newDigits.length === 4) {
+                setCurrentDigits(newDigits);
+            }
+        }
+    }, [transponderCode]);
```

Comment:
This useEffect? It's just gone now! No reason to synchronize vars any more. When the SimVar changes and we're not editing it'll automatically be applied. If we're editing it will be silently updated in the background without affecting our "draft" digits, and if the "draft" is then discarded (e.g. through timeout) we automatically switch to the new version, staying fully in sync with the SimVar.

```suggestion
```